### ISSUE TITLE
[ERROR] CatalogController system-library [catalog] failed with : Clone failed: Cloning into 'management-state/catalog-cache/'... fatal: unable to access 'https://git.rancher.io/system-charts/': Received HTTP code 407 from proxy after CONNECT

### ISSUE THREAD

**Issue by gknepper** [rancher/rancher#20979]
**What kind of request is this (question/bug/enhancement/feature request):**

**Steps to reproduce (least amount of steps as possible):**
I'm trying to install rancher behind a proxy with authentication and SSL offloading and when I ran the command to deploy rancher on a RKE cluster I notice this errors in some of rancher containers:

**helm command to install rancher:**
./helm install rancher-latest/rancher \
  --name rancher \
  --namespace cattle-system \
  --set hostname=rancher-dev.icbc.net \
  --set ingress.tls.source=secret \
  --set proxy="proxystring" \
  --set noProxy="127.0.0.0/8\,10.0.0.0/8\,172.17.0.0/16\,192.168.0.0/16" \
  --set tls=external \
  --set privateCA=true

**Result:**

**Error found in some rancher container logs:**
2019/06/19 16:02:42 [ERROR] CatalogController system-library [catalog] failed with : Clone failed: Cloning into 'management-state/catalog-cache/2284e9d6c1e82c5a56a5cb7c1d5c0144835734b1b66da6f75122d99c7989618f'...
fatal: unable to access 'https://git.rancher.io/system-charts/': Received HTTP code 407 from proxy after CONNECT
: exit status 128

**Other details that may be helpful:**

**When I enter in the container I notice the variables are set:**

root@rancher-7794977df8-wwhzn:/var/lib/rancher# env  | grep _PROXY
HTTP_PROXY=proxystring
NO_PROXY=127.0.0.0/8,10.0.0.0/8,172.17.0.0/16,192.168.0.0/16
HTTPS_PROXY=proxystring

**But git is not working:**

root@rancher-7794977df8-wwhzn:~# git ls-remote https://git.rancher.io/charts/
fatal: unable to access 'https://git.rancher.io/charts/': Received HTTP code 407 from proxy after CONNECT

Eventually the rancher console never get started and the installation keep stuck. 

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
- Installation option (HA):

docker.io/rancher/rancher                                   v2.2.4 

**Cluster information**

- Cluster type (Hosted):
- Machine type (VM) and specifications (4 CPU/8GB RAM):
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.2", GitCommit:"66049e3b21efe110454d67df4fa62b08ea79a19b", GitTreeState:"clean", BuildDate:"2019-05-16T16:23:09Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.5", GitCommit:"2166946f41b36dea2c4626f90a77706f426cdea2", GitTreeState:"clean", BuildDate:"2019-03-25T15:19:22Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
Client:
 Version:         1.13.1
 API version:     1.26
 Package version: docker-1.13.1-96.gitb2f74b2.el7.x86_64
 Go version:      go1.10.8
 Git commit:      b2f74b2/1.13.1
 Built:           Tue Apr  2 21:01:07 2019
 OS/Arch:         linux/amd64

Server:
 Version:         1.13.1
 API version:     1.26 (minimum version 1.12)
 Package version: docker-1.13.1-96.gitb2f74b2.el7.x86_64
 Go version:      go1.10.8
 Git commit:      b2f74b2/1.13.1
 Built:           Tue Apr  2 21:01:07 2019
 OS/Arch:         linux/amd64
 Experimental:    false
```

**Comment by sg0r4n** [comment]
Hi, try setting http_proxy and https_proxy info also for the nodes. It worked for me.

**Comment by gknepper** [comment]
Hi there, yes the proxy is set on nodes too. I think this was a bug in version v2.2.3, mostly because I installed the version v2.2.4 and the problem is gone.

**Comment by nuaaparnec** [comment]
@gknepper Did you have this problem in the v2.2.4,I also have in this error during my installation for v2.2.4. Could you pls share me some info

**Comment by nuaaparnec** [comment]
I try to install the rancher(2.2.4) in HA。 helm command is :
helm install ./rancher \
    --name rancher \
    --namespace cattle-system \
    --set hostname=rancher.ctyun.com \
    --set ingress.tls.source=secret \
    --set auditLog.level=3 \
    --set auditLog.destination=hostPath \
    --set auditLog.hostPath=/opt/rancher/auditlogs/ \
    --set auditLog.maxAge=20 \
    --set auditLog.maxBackups=20 \
    --set auditLog.maxSize=100 \
    --set privateCA=true \
    --set rancherImage=rancher/rancher
I have set up the nginx in front of the cluster.
However the cluster cannot be starting up.
The error logs below:
2019/07/22 10:27:47 [DEBUG] REST UPDATE apis/management.cattle.io/v3//catalogs/system-library
2019/07/22 10:27:47 [ERROR] CatalogController system-library [catalog] failed with : Clone failed: Cloning into 'management-state/catalog-cache/2284e9d6c1e82c5a56a5cb7c1d5c0144835734b1b66da6f75122d99c7989618f'...
fatal: unable to access 'https://git.rancher.io/system-charts/': Could not resolve host: git.rancher.io
: exit status 128
2019/07/22 10:27:49 [INFO] Received SIGTERM, cancelling
2019/07/22 10:27:49 [FATAL] leaderelection lost for cattle-controllers

**Comment by gknepper** [comment]
@nuaaparnec Your error it is not the same as mine. Your's look more like a DNS/proxy networking issue. Mine was solved in the version 2.2.4.
